node/mock-test.ts

```diff
@@ -125,4 +129,176 @@ export class MockTestRunner {
             console.log('TRACE FILE: ' + traceFile);
         }
     }
+
+    // Returns a path to node.exe with the correct version for this task (based on if its node10 or node)
+    private getNodePath(nodeVersion?: number): string {
+        const version: number = nodeVersion || this.getNodeVersion();
+
+        let downloadVersion: string;
+        switch (version) {
+            case 5:
+                downloadVersion = 'v5.10.1';
+                break;
+            case 6:
+                downloadVersion = 'v6.10.3';
+                break;
+            case 10:
+                downloadVersion = 'v10.15.1';
+                break;
+            default:
+                throw new Error('Invalid node version, must be 5, 6, or 10 (received ' + version + ')');
+        }
+
+        // Install node in home directory if it isn't already there.
+        const downloadDestination: string = path.join(downloadDirectory, 'node' + version);
+        const pathToExe: string = this.getPathToNodeExe(downloadVersion, downloadDestination);
+        if (pathToExe) {
+            return pathToExe;
+        }
+        else {
+            return this.downloadNode(downloadVersion, downloadDestination);
+        }
+    }
+
+    // Determines the correct version of node to use based on the contents of the task's task.json. Defaults to Node 10.
+    private getNodeVersion(): number {
+        const taskJsonPath: string = this.getTaskJsonPath();
+        if (!taskJsonPath) {
+            console.warn('Unable to find task.json, defaulting to use Node 10');
+            return 10;
+        }
+        const taskJson: object = require(taskJsonPath);
+
+        const execution: object = taskJson['execution'];
+        let nodeVersion: number = 0;
+        Object.keys(execution).forEach((key: string) => {
+            if (key.toLowerCase() == 'node') {
+                nodeVersion = 6;
+            }
+            else if (key.toLowerCase() == 'node10') {
```

Comment:
should the order of the if conditions be switched? in other words, prefer 10 over 6?

Reply:
Another thing we could do here that would be helpful, is check for a min agent version in the task.json. Node 6 wasn't added to the agent until 2.117.0, per [this document](https://github.com/Microsoft/azure-pipelines-task-lib/blob/master/node/docs/minagent.md#task-execution-handler)

Reply:
Decided to leave out node 5 check per our offline conversation, do now prefer node 10 though.